azure/scope/identity.go

```diff
@@ -196,12 +181,39 @@ func (p *AzureCredentialsProvider) GetTokenCredential(ctx context.Context, resou
 		}
 		cred, authErr = azidentity.NewClientSecretCredential(p.GetTenantID(), p.Identity.Spec.ClientID, clientSecret, &options)
 
+	case infrav1.ServicePrincipal:
+		log.Info("Identity type ServicePrincipal is deprecated and will be removed in a future release.")
```

Comment:
is it deprecated? I think only `ManualServicePrincipal` should be considered deprecated? We're going to have to keep support for non-workload identity paths at least for the near term

Reply:
If there's no difference between the two now, should we still deprecate one or the other to simplify things? In the [current docs](https://capz.sigs.k8s.io/topics/identities#manual-service-principal-identity), we have `ManualServicePrincipal` as a still-supported mechanism. 

I deprecated `ServicePrincipal` because that seemed like the least amount of change, but we should do whatever is easiest for the user to understand.

Reply:
I agree in practice we should keep both but mark one as deprecated since they are now essentially redundant. `ManualServicePrincipal` seems more wordy to me and might be misleading in the future (what's manual about it?) which is why I'm leaning for keeping the simpler `ServicePrincipal` as the SP option, even if it's a bit more friction to essentially switch the deprecation notice in the near term.

@dtzar do you have any thoughts on this?

Reply:
Agreed on both points: it's a bit more friction to deprecate `ManualServicePrincipal`, but semantically it makes sense to retire that somewhat confusing label and keep `ServicePrincipal`.

Reply:
I like keeping `ServicePrincipal` as a supported option and ditching the `ManualServicePrincipal` name.